packages/tokyo-reusable-crowdsale/contracts/crowdsale/StagedCrowdsale.sol

```diff
@@ -67,15 +68,18 @@ contract StagedCrowdsale is KYCCrowdsale {
         weiRaised: 0
       }));
     }
-
-    /* require(validPeriods()); */
   }
 
   function validPeriods() internal view returns (bool) {
     if (periods.length != numPeriods) {
       return false;
     }
 
+    for (uint8 i = 0; i < periods.length; i++) {
+      if (periods[i].endTime <= periods[i].startTime) {
+        return false;
+      }
+    }
```

Comment:
this can be executed inside of the for statement embracing `periods.push(...)` to reduce additional for statement.